controllers/certificatesigningrequests_controller.go

```diff
@@ -88,6 +88,14 @@ func (r *certificateSigningRequestsReconciler) Reconcile(ctx context.Context, re
 
 // reconcileCSR handles the CSR validation and approval
 func (r *certificateSigningRequestsReconciler) reconcileCSR(request *certificates.CertificateSigningRequest) error {
+	// If a CSR is approved/denied after being added to the queue, but before we reconcile it,
+	// trying to approve it again will result in an error and cause a loop.
+	// Return early if the CSR has been approved/denied externally.
+	if !isPending(request) {
+		r.log.Info("CSR is already approved/denied", "Name", request.Name)
```

Comment:
I think this will log redundant info due to: https://github.com/openshift/windows-machine-config-operator/blob/1d30a80a164ceff6aa0aae75713955c2d4cab186/controllers/certificatesigningrequests_controller.go#L72

Seems like `"Name", request.Name` can be dropped?

Reply:
Looks like this at the moment, from the logs:
`2021-09-14T11:14:03.874Z	INFO	controllers.CertificateSigningRequests	CSR is already approved/denied	{"Name": "csr-6nl2c"}`

Reply:
I think that line is wrong, it was my understanding that when you use `WithValues` it creates a child logger that inherits the values from the parent, not that it modifies the logger in place. I think you want to do something like the below if you want to attach the namespaced name to all log lines

```
requestLogger := r.log.WithValues("certificatesigningrequests", req.NamespacedName) 
```
Then all logging within that function comes via that logger: `requestLogger.Info(...)`

Reply:
+1 @JoelSpeed, [docs](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/log#DelegatingLogger.WithValues):
`WithValues provides a new Logger with the tags appended.`

Reply:
I can add a follow-up PR to fix this.